test/ipcAPI.cpp

```diff
@@ -0,0 +1,342 @@
+        allocations_write_lock_type lock(alloc_mutex);
+        allocations.erase(ptr);
+        lock.unlock();
+        auto ret = helper_prov.free(ptr, size);
+        return ret;
+    }
+    const char *get_name() noexcept { return "mock_ipc"; }
+    enum umf_result_t get_ipc_handle_size(size_t *size) noexcept {
+        *size = sizeof(ipc_data);
+        return UMF_RESULT_SUCCESS;
+    }
+    enum umf_result_t get_ipc_handle(const void *ptr, size_t size,
+                                     void *ipcDataOpaque) noexcept {
+        ++stat->getCount;
+        ipc_data *ipcData = static_cast<ipc_data *>(ipcDataOpaque);
+        allocations_read_lock_type lock(alloc_mutex);
+        auto it = allocations.find(ptr);
+        if (it == allocations.end()) {
+            // client tries to get handle for the pointer that does not match
+            // with any of the base addresses allocated by the instance of
+            // the memory provider
+            return UMF_RESULT_ERROR_INVALID_ARGUMENT;
+        }
+        (void)size;
+        ipcData->ptr = ptr;
+        ipcData->size = it->second; // size of the base allocation
+        return UMF_RESULT_SUCCESS;
+    }
+    enum umf_result_t put_ipc_handle(void *ipcData) noexcept {
+        ++stat->putCount;
+        (void)ipcData;
+        return UMF_RESULT_SUCCESS;
+    }
+    enum umf_result_t open_ipc_handle(void *ipcDataOpaque,
+                                      void **ptr) noexcept {
+        ++stat->openCount;
+        ipc_data *ipcData = static_cast<ipc_data *>(ipcDataOpaque);
+        void *mapping = std::malloc(ipcData->size);
+        if (!mapping) {
+            return UMF_RESULT_ERROR_OUT_OF_HOST_MEMORY;
+        }
+
+        memcpy(mapping, ipcData->ptr, ipcData->size);
+
+        *ptr = mapping;
+
+        return UMF_RESULT_SUCCESS;
+    }
+    enum umf_result_t close_ipc_handle(void *ptr) noexcept {
+        ++stat->closeCount;
+        std::free(ptr);
+        return UMF_RESULT_SUCCESS;
+    }
+};
+
+struct umfIpcTest : umf_test::test {
+    umfIpcTest() : pool(nullptr, nullptr) {}
+    void SetUp() override {
+        test::SetUp();
+        this->pool = makePool();
+    }
+
+    void TearDown() override { test::TearDown(); }
+
+    umf::pool_unique_handle_t makePool() {
+        // TODO: The function is similar to poolCreateExt function
+        //       from memoryPool.hpp
+        umf_memory_provider_handle_t hProvider;
+        umf_memory_pool_handle_t hPool;
+
+        auto ret =
+            umfMemoryProviderCreate(&IPC_MOCK_PROVIDER_OPS, &stat, &hProvider);
+        EXPECT_EQ(ret, UMF_RESULT_SUCCESS);
+
+        ret = umfPoolCreate(umfProxyPoolOps(), hProvider, nullptr,
+                            UMF_POOL_CREATE_FLAG_OWN_PROVIDER, &hPool);
+        EXPECT_EQ(ret, UMF_RESULT_SUCCESS);
+
+        // capture provider and destroy it after the pool is destroyed
+        auto poolDestructor = [](umf_memory_pool_handle_t pool) {
```

Comment:
you don;t need this `poolDestructor` anymore, just pass umfPoolDestroy as a second arg to `pool_unique_handle_t`